mongo/options/clientoptions.go

```diff
@@ -104,6 +105,7 @@ type ClientOptions struct {
 	DisableOCSPEndpointCheck *bool
 	HeartbeatInterval        *time.Duration
 	Hosts                    []string
+	HTTPClient               *http.Client
```

Comment:
What is the use case for supplying a custom HTTP client for a `mongo.Client`?

Reply:
because it gets passed down to the many other nested options, for the same reason why some other options are here :)